sql-statements/sql-statement-create-index.md

```diff
@@ -104,44 +104,108 @@ Query OK, 0 rows affected (0.31 sec)
+>
+> - volatile functions, such as `rand()`, `now()`, and so on.
+> - system variables and user variables.
+> - a subquery.
+> - `AUTO_INCREMENT` column. However, there is one exception: you can remove this restriction by setting the value of `tidb_enable_auto_increment_in_generated` (system variable) to `true`.
+> - window functions.
+> - the ROW functions, such as `create table t (j json, key k (((j,j))));`.
+> - aggregate functions.
+> 
+> An expression index takes names implicitly by setting `_V$_{index_name}_{index_offset}`. Suppose you try to set a new expression index's name that a column has taken when creating an expression index. In that case, an error returns in this process. Besides, if you add a column with an existing name, the same error returns.
+>
+> The number of parameters of the functions that are in an expression with expression indexes must be correct.
 
-To use this feature, make the following setting in [TiDB Configuration File](/tidb-configuration-file.md#allow-expression-index-new-in-v400):
+When the expression in a query statement matches the expression in an expression index, the optimizer can choose to use the expression index for the query. In some cases, the optimizer may not select an expression index depending on statistics. At that time, you can force the optimizer to select an expression index by specifying optimizer hints.
+
+For example, suppose you build the expression index `idx` on the expression `lower(col1)`:
+
+If the results of the query statement are the same expressions, you can use expression indexes. Take the following statements as an example:
 
 {{< copyable "sql" >}}
 
 ```sql
-allow-expression-index = true
+SELECT lower(col1) FROM t;
 ```
 
-TiDB can build indexes not only on one or more columns in a table, but also on an expression. When queries involve expressions, expression indexes can speed up those queries.
+If the same expressions are included in the filtering conditions, you can use expression indexes. Take the following statements as an example:
 
-Take the following query as an example:
+{{< copyable "sql" >}}
+
+```sql
+SELECT * FROM t WHERE lower(col1) = "a";
+SELECT * FROM t WHERE lower(col1) > "a";
+SELECT * FROM t WHERE lower(col1) BETWEEN "a" AND "b";
+SELECT * FROM t WHERE lower(col1) in ("a", "b");
+SELECT * FROM t WHERE lower(col1) > "a" AND lower(col1) < "b";
+SELECT * FROM t WHERE lower(col1) > "b" OR lower(col1) < "a";
+```
+
+When the queries are sorted by the same expressions, you can use expression indexes. Take the following statements as an example:
 
 {{< copyable "sql" >}}
 
 ```sql
-SELECT * FROM t WHERE lower(name) = "pingcap";
+SELECT * FROM t ORDER BY lower(col1);
 ```
 
-If the following expression index is built, you can use the index to speed up the above query:
+If the same expressions are included in the aggregate (`GROUP BY`) functions, you can use expression indexes. Take the following statements as an example:
```

Comment:
Please modify this sentence according to the above comments. Thanks

Reply:
```suggestion
If the same expression is included in the aggregate (`GROUP BY`) functions, the expression index applies. Take the following statement as an example:
```